### PR TITLE
feat: allow categories to display all child documents

### PR DESCRIPTION
**Pull request by Yelinz**
closes https://github.com/projectcaluma/ember-alexandria/issues/159

### CODE REVIEW

File: addon/components/document-view.js [hunk 1]
```diff
@@ -51,9 +51,21 @@ export default class DocumentViewComponent extends Component {
...
+    if (
+      getOwnConfig().fetchChildCategoryDocuments?.includes(
+        this.args.filters.category,
+      )
+    ) {
+      filter.categories = this.store
+        .peekRecord("category", this.args.filters.category)
+        .children.map((category) => category.id);
+      delete filter.category;
+    }
```
Comment (czosel) [inline addon/components/document-view.js:64]: Let's rename the filter to `categories` and always filter for the selected category plus all its children.

File: ember-cli-build.js [hunk 2]
```diff
@@ -7,6 +7,9 @@ module.exports = function (defaults) {
     "ember-simple-auth": {
       useSessionSetupMethod: true,
     },
+    "ember-alexandria": {
+      fetchChildCategoryDocuments: ["beilagen-zum-gesuch"],
+    },
```
Comment (czosel) [inline ember-cli-build.js:12]: can be dropped

File: index.js [hunk 3]
```diff
@@ -9,4 +9,23 @@ module.exports = buildEngine({
...
+  included(...args) {
+    this._super.included.apply(this, ...args);
+
+    const app = this._findHost(this);
+
+    this.options.options[
+      "@embroider/macros"
+    ].setOwnConfig.fetchChildCategoryDocuments =
+      app.options["ember-alexandria"]?.fetchChildCategoryDocuments ?? [];
+  },
```
Comment (czosel) [inline index.js:30]: can be dropped as well

File: addon/components/document-view.js (hunk 1 above)
Comment (Yelinz) [reply addon/components/document-view.js:65]: Done

File: ember-cli-build.js (hunk 2 above)
Comment (Yelinz) [reply ember-cli-build.js]: Done

**Comment by anehx** [comment]
Huh, I would've just implemented this as the default behaviour of the category filter in the backend. Is there a reason we do this in the frontend?

**Comment by anehx** [comment]
@Yelinz I think we can close this because of https://github.com/projectcaluma/alexandria/pull/365 rigth?

**Comment by Yelinz** [comment]
Yes, this has become obsolete

**Pull request closed without merging**